hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleContextImpl.java

```diff
@@ -584,6 +597,66 @@ private void dispatchSyntheticTxn(
         }
     }
 
+    private record ValidationResult(
+            @NonNull PreHandleResult.Status status, @NonNull ResponseCodeEnum responseCodeEnum) {}
+
+    private ValidationResult validate(
+            @NonNull final HandleContextVerifier verifier,
+            HederaFunctionality function,
+            TransactionBody txBody,
+            AccountID payer,
+            Key payerKey) {
+
+        PreHandleContextImpl preHandleContext;
+        try {
+            preHandleContext =
+                    new PreHandleContextImpl(readableStoreFactory(), txBody, payer, configuration(), dispatcher);
+            dispatcher.dispatchPreHandle(preHandleContext);
+        } catch (PreCheckException e) {
+            return new ValidationResult(PRE_HANDLE_FAILURE, e.responseCode());
+        }
+
+        // Check if the payer has the required permissions
+        if (!authorizer.isAuthorized(payer, function)) {
+            if (function == HederaFunctionality.SYSTEM_DELETE) {
+                return new ValidationResult(PRE_HANDLE_FAILURE, ResponseCodeEnum.NOT_SUPPORTED);
+            }
+            return new ValidationResult(PRE_HANDLE_FAILURE, ResponseCodeEnum.UNAUTHORIZED);
+        }
+
+        // Check if the transaction is privileged and if the payer has the required privileges
+        final var privileges = authorizer.hasPrivilegedAuthorization(payer, function, txBody);
+        if (privileges == SystemPrivilege.UNAUTHORIZED) {
+            return new ValidationResult(PRE_HANDLE_FAILURE, ResponseCodeEnum.AUTHORIZATION_FAILED);
+        }
+        if (privileges == SystemPrivilege.IMPERMISSIBLE) {
+            return new ValidationResult(PRE_HANDLE_FAILURE, ResponseCodeEnum.ENTITY_NOT_ALLOWED_TO_DELETE);
+        }
+
+        // Check all signature verifications. This will also wait, if validation is still ongoing.
+        final var payerKeyVerification = verifier.verificationFor(payerKey);
```

Comment:
We need an option here to skip verification of the payer key when dispatching EVM child transactions.

If we don't have this option, then consider:
 - A contract `0.0.C` has a cryptographic admin key `K`.
 - A user calls contract `0.0.C`, and it e.g. calls the `tokenTransfer()` system contract to transfer some tokens it controls.

Now, we do **NOT** want to require key `K` signed the top-level HAPI `ContractCall`; because authorization in this case is granted by the logic in the smart contract itself, and not a HAPI signature.